extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -431,6 +476,332 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
+		await valueGroup.updateCssStyles({ 'margin-left': '-10px' });
+		return valueGroup;
+	}
+
+	/**
+	 * Make the dropdowns section for the selected database scoped configuration visible
+	 * @param isSecondaryCheckboxChecked - Whether the secondary checkbox is checked or not
+	 */
+	private async showDropdownsSection(isSecondaryCheckboxChecked: boolean): Promise<void> {
+		this.setSecondaryCheckboxForDropdowns.checked = isSecondaryCheckboxChecked;
+		await this.dscPrimaryValueDropdownGroup.updateCssStyles({ 'visibility': 'visible' });
+		await this.dscSecondaryCheckboxForDropdownGroup.updateCssStyles({ 'visibility': 'visible' });
+		await this.dscSecondaryValueDropdownGroup.updateCssStyles({ 'visibility': isSecondaryCheckboxChecked ? 'hidden' : 'visible' });
+	}
+
+	/**
+	 * Make the input section for the selected database scoped configuration visible
+	 * @param isSecondaryCheckboxChecked - Whether the secondary checkbox is checked or not
+	 */
+	private async showInputSection(isSecondaryCheckboxChecked: boolean): Promise<void> {
+		this.setSecondaryCheckboxForInputType.checked = isSecondaryCheckboxChecked;
+		await this.dscPrimaryValueInputGroup.updateCssStyles({ 'visibility': 'visible', 'margin-top': '-175px' });
+		await this.dscSecondaryCheckboxForInputGroup.updateCssStyles({ 'visibility': 'visible', 'margin-top': '-120px' });
+		await this.dscSecondaryValueInputGroup.updateCssStyles({ 'visibility': isSecondaryCheckboxChecked ? 'hidden' : 'visible', 'margin-top': '-85px' });
+	}
+
+	/**
+	 * Set all primary and secondary groups to hidden
+	 */
+	private async hideDropdownAndInputSections(): Promise<void> {
+		await this.dscPrimaryValueInputGroup.updateCssStyles({ 'visibility': 'hidden', 'margin-top': '0px' });
+		await this.dscSecondaryCheckboxForInputGroup.updateCssStyles({ 'visibility': 'hidden', 'margin-top': '0px' });
+		await this.dscSecondaryValueInputGroup.updateCssStyles({ 'visibility': 'hidden', 'margin-top': '0px' });
+		await this.dscPrimaryValueDropdownGroup.updateCssStyles({ 'visibility': 'hidden' });
+		await this.dscSecondaryCheckboxForDropdownGroup.updateCssStyles({ 'visibility': 'hidden' });
+		await this.dscSecondaryValueDropdownGroup.updateCssStyles({ 'visibility': 'hidden' });
+	}
+
+	/**
+	 * Updates the data to the table and sets the focus to the selected row
+	 * @param data - Modified data to be set in the table
+	 */
+	private async updateDscTable(data: any[][], needFocus: string = null): Promise<void> {
+		// Set the focus to the selected input box
+		this.setFocusToInput = needFocus;
+		await this.setTableData(this.dscTable, data, DscTableRowLength);
+		// Restore the focus to previously selected row.
+		this.dscTable.setActiveCell(this.currentRowId, 0);
+	}
+
+	/**
+	 * Input box types requires focus to be set after refreshing the table data
+	 */
+	private async setFocusToInputBox(): Promise<void> {
+		if (this.setFocusToInput === 'primaryInput') {
```

Comment:
Would suggest just having this var be the actual component to focus instead - and then undefined if nothing to focus